serverlessgo/main.go

```diff
@@ -15,25 +15,66 @@ var (
 	lambda   = flag.Float64("lambda", 140.0, "The lambda of the Poisson distribution used on workload.")
 )
 
+var arrivalQueue = godes.NewFIFOQueue("arrival")
+var arrivalCond = godes.NewBooleanControl()
+
 func main() {
-	fmt.Println("Simulation Started")
 	flag.Parse()
 	
-	arrivalQueue := godes.NewFIFOQueue("arrival")
+	lb := newLoadBalancer()
+	godes.AddRunner(lb)
+
 	poissonDist := &distuv.Poisson{*lambda, rand.NewSource(uint64(time.Now().Nanosecond()))}
 	reqID := int64(0)
 	godes.Run()
 	for godes.GetSystemTime() < duration.Seconds() {
 		arrivalQueue.Place(&request{id: reqID})
+		arrivalCond.Set(true)
 		interArrivalTime := poissonDist.Rand()
 		godes.Advance(interArrivalTime)
-		r := arrivalQueue.Get().(*request)
-		r.responseTime = interArrivalTime 
 		reqID++
 	}
 
+	lb.terminate()
 	godes.WaitUntilDone()
-	fmt.Println("Simulation Ended")
+}
+
+type loadBalancer struct {
+	*godes.Runner
+	isTerminated bool
+}
+
+func newLoadBalancer() *loadBalancer {
+	return &loadBalancer{&godes.Runner{}, false}
+}
+
+func (lb *loadBalancer) terminate() {
+	arrivalCond.Set(true)
+	lb.isTerminated = true
+}
+
+func (lb *loadBalancer) Run() {
+	fmt.Println("timestamp,id,status,latency")
+	for {
+		arrivalCond.Wait(true)
+		if arrivalQueue.Len() > 0 {
+			r := arrivalQueue.Get().(*request)
+			if r.status != 200 {
```

Comment:
Como conversamos, simplesmente ignore quando não for 200 nessa versão.

Reply:
Beleza